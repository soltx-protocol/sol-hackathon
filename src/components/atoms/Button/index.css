:root {
	--button-normal: var(--primary-cta);
	--button-normal-hover: var(--primary-cta-hover);
	--button-normal-active: var(--primary-cta-active);
	--button-normal-disabled: var(--primary-cta-disabled);

	/* outline */
	--button-outline: var(--neutral-control-layer);
	--button-outline-hover: var(--neutral-control-layer-hover);
	--button-outline-active: var(--neutral-control-layer-active);
	--button-outline-disabled: var(--neutral-control-layer-disabled);
	--button-outline-border: var(--neutral-control-70);
	--button-outline-border-hover: var(--neutral-control-90);
	--button-outline-border-active: var(--neutral-control-70);
	--button-outline-border-disabled: var(--neutral-control-disabled);
}

.button {
	display: inline-flex;
	min-width: 109px;
	height: 38px;
	padding: 9px 17px;
	border-radius: 6px;
	background-color: var(--button-normal);
	box-shadow: 0 1px 2px rgb(0 0 0 / 5%);
	border: solid 1px var(--button-normal);
	transition: border-color 0.3s ease-out;
	transition: background-color 0.3s ease-out;

	& > div {
		display: flex;
		width: 100%;
		height: 100%;
		align-items: center;
		justify-content: center;
		font-style: normal;
		font-weight: 500;
		font-size: 14px;
		line-height: 20px;
		text-align: center;
		color: var(--neutral-white);
		transition: color 0.3s ease-out;

		& > svg {
			width: 20px;
			height: 20px;
			margin-left: 8px;

			& path {
				fill: var(--neutral-white);
			}
		}
	}

	&.large {
		min-width: 120px;
		height: 42px;

		& > div {
			font-weight: 500;
			font-size: 16px;
			line-height: 24px;
		}
	}

	&.small {
		min-width: 101px;
		height: 34px;
		padding: 9px 13px;

		& > div {
			line-height: 16px;
		}
	}

	&:hover {
		background-color: var(--button-normal-hover);
		border-color: var(--button-normal-hover);
	}

	&:active,
	&.active {
		background-color: var(--button-normal-active);
		border-color: var(--button-normal-active);
	}

	&.disabled {
		background-color: var(--button-normal-disabled);
		border-color: var(--button-normal-disabled);
		cursor: not-allowed;
	}

	&.outline {
		background-color: var(--neutral-white);
		border-color: var(--button-outline-border);

		&.transparent {
			background-color: transparent;
		}

		& > div {
			color: var(--button-outline);

			& > svg path {
				fill: var(--button-outline);
			}
		}

		&:hover {
			& > div {
				color: var(--button-outline-hover);

				& > svg path {
					fill: var(--button-outline-hover);
				}
			}
		}

		&:active,
		&.active {
			border-color: var(--button-outline-border-active);

			/* stylelint-disable-next-line no-descending-specificity */
			& > div {
				color: var(--button-outline-active);

				/* stylelint-disable-next-line no-descending-specificity */
				& > svg path {
					fill: var(--button-outline-active);
				}
			}
		}

		&.disabled {
			border-color: var(--button-outline-border-disabled);
			cursor: not-allowed;

			/* stylelint-disable-next-line no-descending-specificity */
			& > div {
				color: var(--button-outline-disabled);

				/* stylelint-disable-next-line no-descending-specificity */
				& > svg path {
					fill: var(--button-outline-disabled);
				}
			}
		}
	}
}
