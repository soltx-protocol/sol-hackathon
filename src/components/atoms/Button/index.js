/* eslint-disable react/jsx-props-no-spreading */

import React from 'react';
import classnames from 'classnames';

// import { isExist } from 'util/helper';
import { isExist } from 'util/helper';

import styles from './index.css';

export const BUTTON_SIZE_TYPE = {
	LARGE: 'LARGE',
	MEDIUM: 'MEDIUM',
	SMALL: 'SMALL',
};

const Button = ({
	children,
	className,
	disabled = false,
	outline = false,
	transparent = false,
	active = false,
	onClick = () => {},
	Icon,
	size = BUTTON_SIZE_TYPE.MEDIUM,
	...other
}) => (
	<button
		className={classnames(styles.button, className, {
			[styles.disabled]: disabled,
			[styles.icon]: isExist(Icon),
			[styles.outline]: outline,
			[styles.transparent]: transparent,
			[styles.active]: active,
			[styles.large]: size === BUTTON_SIZE_TYPE.LARGE,
			[styles.small]: size === BUTTON_SIZE_TYPE.SMALL,
		})}
		type="button"
		onClick={e => {
			if (!disabled) {
				onClick(e);
			}
		}}
		{...other}
	>
		<div>
			{children}
			{Icon && <Icon className={styles.icon} />}
		</div>
	</button>
);

export default Button;
