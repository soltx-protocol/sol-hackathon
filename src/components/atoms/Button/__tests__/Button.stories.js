import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, select } from '@storybook/addon-knobs/react';

import Button, { BUTTON_SIZE_TYPE } from 'components/atoms/Button';

import MailIcon from 'images/icon/mail.inline.svg';

const stories = storiesOf('atoms/Button', module);

stories.add('__interactive', () => (
	<Button
		outline={boolean('outline', false)}
		disabled={boolean('disabled', false)}
		active={boolean('active', false)}
		size={select('size', Object.keys(BUTTON_SIZE_TYPE), BUTTON_SIZE_TYPE.MEDIUM)}
	>
		瞭解更多
	</Button>
));

stories.add('with Icon', () => (
	<Button Icon={MailIcon} iconRight={boolean('icon right', false)}>
		瞭解更多
	</Button>
));
