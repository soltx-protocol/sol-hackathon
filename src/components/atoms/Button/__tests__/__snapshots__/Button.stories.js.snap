// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="button"
  >
    <div>
      瞭解更多
    </div>
  </button>
</DocumentFragment>
`;
